### DISCUSSION TITLE
Upgraded from 2.1.0 to 3.6.3 having issues can someone assist? ModbusTcpClient

### DISCUSSION THREAD

**Discussion by optio50** [pymodbus-dev/pymodbus#1951] [Q&A]
This code works on 2.1.0 using Ubuntu 22.04.3 LTS
```
from pymodbus.constants import Defaults
from pymodbus.constants import Endian
from pymodbus.client.sync import ModbusTcpClient as ModbusClient
from pymodbus.payload import BinaryPayloadDecoder

BmvID = 223

ip = "192.168.20.156"

client = ModbusClient(ip, port='502')

def modbus_register(address, unit):
    msg     = client.read_holding_registers(address, unit=unit)
    decoder = BinaryPayloadDecoder.fromRegisters(msg.registers, byteorder=Endian.Big)
    msg     = decoder.decode_16bit_int()
    return msg

BatterySOC  = modbus_register(266, BmvID) / 10

print(f"BatterySOC {BatterySOC}")

```

after modifying what I thought might work for 3.6.3
```
from pymodbus.constants import Endian
from pymodbus.client import ModbusTcpClient as ModbusClient
from pymodbus.payload import BinaryPayloadDecoder

BmvID = 223

ip = "192.168.20.156"

client = ModbusClient(ip, port='502')

def modbus_register(address, unit):
    msg     = client.read_holding_registers(address, unit=unit)
    decoder = BinaryPayloadDecoder.fromRegisters(msg.registers, byteorder=Endian.BIG)
    msg     = decoder.decode_16bit_int()
    return msg

BatterySOC  = modbus_register(266, BmvID) / 10

print(f"BatterySOC {BatterySOC}")
```

I receive the error

`Traceback (most recent call last):
  File "/home/chromebox/Scripts/Python/./battery.py", line 19, in <module>
    BatterySOC  = modbus_register(266, BmvID) / 10
  File "/home/chromebox/Scripts/Python/./battery.py", line 16, in modbus_register
    msg     = decoder.decode_16bit_int()
  File "/home/chromebox/.local/lib/python3.10/site-packages/pymodbus/payload.py", line 406, in decode_16bit_int
    return unpack(fstring, handle)[0]
struct.error: unpack requires a buffer of 2 bytes`

**Comment by optio50** [comment]
found the issue.
unit is no longer a valid parameter. Replace with slave

```
def modbus_register(address, unit):
    msg     = client.read_holding_registers(address, slave=unit)
    decoder = BinaryPayloadDecoder.fromRegisters(msg.registers, byteorder=Endian.BIG)
    msg     = decoder.decode_16bit_int()
    return msg
```

**Comment by janiversen** [comment]
Please read the documentation, you have not upgraded your code to the 3.6.3 call standard.

A good way is to look at our examples, like the simple_client ones.

Why do you want to double decode the registers, that is very likely not going to work on little endian CPUs
